File: practice/star_patters.c
#include<stdio.h>

void main(){
int n = 5;

//  STAR PATTERN 1
//  *
//  **
//  ***
//  ****
//  *****
// for (int i = 0; i < n; i++){
//     for(int j = 0; j <= i; j++){
//         printf("*");
//     }
//     printf("\n");
// }


//  STAR PATTERN 2
//      *
//     **
//    ***
//   ****
//  *****
// for (int i = 0; i < n; i++){
//     for (int j = 0; j <= n-i; j++){
//         printf(" ");
//     }
//     for (int k = 0; k <= i; k++){
//         printf("*");
//     }
//     printf("\n");
// }


//  STAR PATTERN 3
//      *
//     * *
//    * * *
//   * * * *
//  * * * * *
for (int i = 0; i < n; i++){
    for (int j = 0; j < n - i; j++){
        printf(" ");
    }
    for (int k = 0; k <= i; k++){
        printf("* ");
    }
    printf("\n");
}


//  extra
//  *****
//  ****
//  ***
//  **
//  *
// for (int i = 0; i < n; i++){
//     for (int k = n-i; k > 0; k--){
//         printf("*");
//     }
//     printf("\n");
// }


//  STAR PATTERN 4
//  *****
//   ****
//    ***
//     **
//      *
// for (int i = 0; i < n; i++){
//     for (int j = 0; j < i; j++){
//         printf(" ");
//     }
//     for (int k = n-i; k > 0; k--){
//         printf("*");
//     }
//     printf("\n");
// }



//  STAR PATTERN 5
//  * * * *
//  *     *
//  *     *
//  * * * *
// for (int i = 0; i < n; i++){
//     for(int j = 0; j < n; j++)
//         if (i == 0 || j == 0 || i == n-1 || j == n-1)
//             printf("*");
//         else
//             printf(" ");
//     printf("\n");
// }
}

File: practice/star_patters.py
# STAR PATTERN 1
# *
# **
# ***
# ****
# *****

# n = 5
# for i in range(0 , n+1):
#     for j in range(0, i):
#         print("*", end="")
#     print()


# STAR PATTERN 2
#     *
#    **
#   ***
#  ****
# *****

# for i in range(0, n+1):
#     for j in range(n-i):
#         print(" ", end="")
#     for k in range(i):
#         print("*", end="")
#     print()


# STAR PATTERN 3
#     *
#    * *
#   * * *
#  * * * *
# * * * * *
# for i in range(0, n+1):
#     for j in range(n-i):
#         print(" ", end="")
#     for k in range(i):
#         print(" *", end="")
#         # print("* ", end="") # both gives same result
#     print()


# extra
# *****
# ****
# ***
# **
# *

# print()
# for i in range(0, n+1):
#     for j in range(n-i):
#         print("*",end="")
#     print()


# STAR PATTERN 4
# *****
#  ****
#   ***
#    **
#     *

# print()
# for i in range(0, n+1):
#     for j in range(i):
#         print(" ",end="")
#     for k in range(n-i):
#         print("*",end="")
#     print()


# STAR PATTERN 5
# * * * *
# *     *
# *     *
# * * * *

# for i in range(n):
#     for j in range(n):
#         if i==0 or i==n-1 or j==0 or j==n-1:
#             print("*",end="")
#         else:
#             print(" ", end="")
#     print()
